test/envtest/kongstate_consumer_failures_test.go

```diff
@@ -0,0 +1,145 @@
+	"k8s.io/client-go/kubernetes/scheme"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/manager"
+	kongv1 "github.com/kong/kubernetes-ingress-controller/v2/pkg/apis/configuration/v1"
+)
+
+func TestKongStateFillConsumersAndCredentialsFailure(t *testing.T) {
+	t.Parallel()
+
+	err := kongv1.AddToScheme(scheme.Scheme)
+	require.NoError(t, err)
+	cfg := Setup(t, scheme.Scheme)
+	InstallKongCRDs(t, cfg)
+	client := NewControllerClient(t, cfg)
+	testIngressClassName := "kong-test"
+
+	// We use a deferred cancel to stop the manager and not wait for its timeout.
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	ns := CreateNamespace(ctx, t, client)
+
+	secrets := []*corev1.Secret{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "key-auth-cred",
+				Namespace: ns.Name,
+			},
+			Data: map[string][]byte{
+				"kongCredType": []byte("key-auth"),
+				"key":          []byte("whatever"),
+				"ttl":          []byte("1024"),
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "empty-cred",
+				Namespace: ns.Name,
+			},
+			Data: map[string][]byte{
+				"kongCredType": []byte("key-auth"),
+			},
+		},
+	}
+
+	kongConsumers := []*kongv1.KongConsumer{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "consumer-key-auth-cred",
+				Namespace: ns.Name,
+				Annotations: map[string]string{
+					"kubernetes.io/ingress.class": testIngressClassName,
+				},
+			},
+			Username: "foo",
+			Credentials: []string{
+				"key-auth-cred",
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "consumer-empty-cred",
+				Namespace: ns.Name,
+				Annotations: map[string]string{
+					"kubernetes.io/ingress.class": testIngressClassName,
+				},
+			},
+			CustomID: "bar",
+			Credentials: []string{
+				"empty-cred",
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "consumer-no-username",
+				Namespace: ns.Name,
+				Annotations: map[string]string{
+					"kubernetes.io/ingress.class": testIngressClassName,
+				},
+			},
+			Credentials: []string{
+				"key-auth-cred",
+			},
+		},
+	}
+
+	for _, secret := range secrets {
+		require.NoError(t, client.Create(ctx, secret))
+	}
+	for _, kongConsumer := range kongConsumers {
+		require.NoError(t, client.Create(ctx, kongConsumer))
+	}
+
+	// KongConsumer name -> event message
+	kongConsumerTranslationFailureMessages := map[string]string{
+		"consumer-empty-cred":  `credential "empty-cred" failure: failed to provision credential: empty secret`,
+		"consumer-no-username": `no username or custom_id specified`,
+	}
+
+	RunManager(ctx, t, cfg, func(cfg *manager.Config) {
+		// Reducing controllers' cache synchronisation timeout in order to trigger the possible sync timeout quicker.
+		// It's a regression test for https://github.com/Kong/gateway-operator/issues/326.
+		cfg.CacheSyncTimeout = time.Millisecond * 500
+		cfg.IngressClassName = testIngressClassName
```

Comment:
I think we could simply use a default ingress class name and prevent custom manager config this way:

```go
Annotations: map[string]string{annotations.IngressClassKey: annotations.DefaultIngressClass},
```

Reply:
done.